### PR TITLE
fix: bump blitzar version to 3.0.2

### PR DESCRIPTION
**Pull request by iajoiner**
# Rationale for this change
We need to check and make sure that [this fix](https://github.com/spaceandtimelabs/blitzar/pull/144) is properly used.

# What changes are included in this PR?
- bump blitzar version to 3.0.2
- add tests to ensure that commitments work on integer extrema
- fix existing tests that once require workarounds

# Are these changes tested?
Yes

### CODE REVIEW

**Comment by iajoiner** [comment]
Please rerun CI when https://github.com/spaceandtimelabs/blitzar-rs/actions/runs/9651262273/job/26618981364 is run

File: crates/proof-of-sql/tests/integration_tests.rs [hunk 1]
```diff
@@ -160,6 +160,81 @@ fn we_can_prove_a_basic_inequality_query_with_curve25519() {
...
+#[test]
+#[cfg(feature = "blitzar")]
+fn we_can_prove_a_basic_query_containing_extrema_with_curve25519() {
+    let mut accessor = OwnedTableTestAccessor::<InnerProductProof>::new_empty_with_setup(());
+    accessor.add_table(
+        "sxt.table".parse().unwrap(),
+        owned_table([
+            smallint("smallint", [i16::MIN, i16::MAX]),
+            int("int", [i32::MIN, i32::MAX]),
+            bigint("bigint", [i64::MIN, i64::MAX]),
```
Comment (drcapybara) [inline crates/proof-of-sql/tests/integration_tests.rs:172]: can we please include zero in these tests, Im also not sure if its worth testing the other numerical boundaries as well such as i128

Comment (drcapybara) [reply crates/proof-of-sql/tests/integration_tests.rs:172]: decimal75 boundaries would be whatever the scalar min max are I suppose

Comment (drcapybara) [reply crates/proof-of-sql/tests/integration_tests.rs:172]:
```suggestion
            smallint("smallint", [i16::MIN, 0, i16::MAX]),
            int("int", [i32::MIN, 0, i32::MAX]),
            bigint("bigint", [i64::MIN, 0, i64::MAX]),
```

**Review by JayWhite2357 — APPROVED** [review]

**Review by drcapybara — APPROVED** [review]
Looks great thanks!
